airflow/models/dag.py

```diff
@@ -2814,10 +2815,24 @@ def deactivate_deleted_dags(cls, alive_dag_filelocs: List[str], session=None):
         """
         log.debug("Deactivating DAGs (for which DAG files are deleted) from %s table ", cls.__tablename__)
 
+        dag_filelocs = []
+        for fileloc in alive_dag_filelocs:
+            if zipfile.is_zipfile(fileloc):
+                with zipfile.ZipFile(fileloc) as z:
+                    dag_filelocs.extend(
```

Comment:
I wonder if this should be done once in `_refresh_dag_dir` instead of done twice here and in `remove_deleted_dags`?

Reply:
Mistake adding it here, it's the `DagCode.remove_deleted_dags` that needed it too. Will move the logic to `_refresh_dag_dir`. Thanks